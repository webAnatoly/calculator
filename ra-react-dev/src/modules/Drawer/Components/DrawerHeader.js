import React, { Component } from 'react'
import Divider from 'material-ui/Divider'
import Avatar from 'material-ui/Avatar'
import AccountCircle from 'material-ui-icons/AccountCircle'
import { ListItem, ListItemText } from 'material-ui/List'
import { withStyles } from 'material-ui/styles'
import { styles } from '../theme'

export class HeaderBlock extends Component {
  render () {
    const { primary, secondary, toggleHiddenBlock, classes } = this.props
    return (
      <div className={classes.drawerHeader}>
        <ListItem onClick={toggleHiddenBlock} button>
          <Avatar>
            <AccountCircle />
          </Avatar>
          <ListItemText
            primary={primary}
            secondary={secondary}
            classes={{
              root: classes.avatarText,
              text: classes.listItemText
            }}
          />
        </ListItem>
        <Divider />
      </div>
    )
  }
}

export default withStyles(styles, { withTheme: true })(HeaderBlock)
