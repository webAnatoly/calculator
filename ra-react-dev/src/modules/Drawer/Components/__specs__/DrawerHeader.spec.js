import React from 'react'
import { shallow } from 'enzyme'
import { HeaderBlock } from '../DrawerHeader'

describe('DrawerHeader', () => {
  let toggleHiddenBlock, primary, secondary, classes, wrapper
  beforeEach(() => {
    toggleHiddenBlock = jest.fn()
    primary = 'primary'
    secondary = 'secondary'
    classes = {
      drawerHeader: 'drawerHeader',
      avatarText: 'avatarText',
      listItemText: 'listItemText'
    }
    wrapper = shallow(
      <HeaderBlock
        classes={classes}
        toggleHiddenBlock={toggleHiddenBlock}
        primary={primary}
        secondary={secondary}
      />
    )
  })

  it('should renders as expected', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should handle onClick()', () => {
    wrapper
      .childAt(0)
      .props()
      .onClick()
    expect(toggleHiddenBlock.mock.calls.length).toEqual(1)
  })

  it('should render ListItemText with primary and secondary text', () => {
    expect(
      wrapper
        .childAt(0)
        .childAt(1)
        .props().primary
    ).toEqual(primary)
    expect(
      wrapper
        .childAt(0)
        .childAt(1)
        .props().secondary
    ).toEqual(secondary)
  })

  it('should render elements with defined classes', () => {
    expect(wrapper.props().className).toEqual(classes.drawerHeader)
    expect(
      wrapper
        .childAt(0)
        .childAt(1)
        .props().classes
    ).toMatchObject({
      root: classes.avatarText,
      text: classes.listItemText
    })
  })
})
