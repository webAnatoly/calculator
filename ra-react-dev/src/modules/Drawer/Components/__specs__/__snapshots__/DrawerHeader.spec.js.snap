// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerHeader should renders as expected 1`] = `
<div
  className="drawerHeader"
>
  <withStyles(ListItem)
    button={true}
    onClick={[Function]}
  >
    <withStyles(Avatar)>
      <pure(AccountCircle) />
    </withStyles(Avatar)>
    <withStyles(ListItemText)
      classes={
        Object {
          "root": "avatarText",
          "text": "listItemText",
        }
      }
      primary="primary"
      secondary="secondary"
    />
  </withStyles(ListItem)>
  <withStyles(Divider) />
</div>
`;
